var/spack/repos/builtin/packages/py-hypothesis/package.py

```diff
@@ -21,9 +21,11 @@ class PyHypothesis(PythonPackage):
     # TODO: Add missing dependency required to import hypothesis.extra.django
 
     version('4.41.2', sha256='6847df3ffb4aa52798621dd007e6b61dbcf2d76c30ba37dc2699720e2c734b7a')
+    version('4.24.3', sha256='fd90a319f409f34a173156ca704d6c0c6c0bb30a2e43dbf26aced2c75569e5d5')
     version('4.7.2',  sha256='87944c6379f77634474b88abbf1e5ed5fe966637cc926131eda5e2af5b54a608')
     version('3.7.0',  sha256='0fea49d08f2d5884f014151a5af6fb48d862f6ad567ffc4a2e84abf2f186c423')
 
     depends_on('py-setuptools', type='build')
-    depends_on('py-attrs@19.2.0:', type=('build', 'run'))
+    depends_on('py-attrs@19.2.0:', type=('build', 'run'), when='@4.41.2:')
+    depends_on('py-attrs@19.1.0:', type=('build', 'run'), when='@4.24.3:')
```

Comment:
4.24.3 requires atrs 16.0.0+, not 19.1.0+. Also, older versions still require attrs. If there's no reason why you can't use attrs 19.2.0+, I think using the least common denominator like we were is fine. If for some reason you need to use an older version of attrs, we'll have to go through the whole history of the package to see when the required version was incremented.

Reply:
`16.0.0+` is fine with me.

Reply:
Also, older versions still require attrs. If there's no reason why you can't use attrs 19.2.0+, I think using the least common denominator like we were is fine. If for some reason you need to use an older version of attrs, we'll have to go through the whole history of the package to see when the required version was incremented.